ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/captureToImageCompat.kt

```diff
@@ -0,0 +1,262 @@
+package com.revenuecat.purchases.ui.revenuecatui.helpers
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.graphics.Bitmap
+import android.graphics.Rect
+import android.os.Build
+import android.os.Handler
+import android.os.Looper
+import android.view.PixelCopy
+import android.view.View
+import android.view.Window
+import androidx.annotation.DoNotInline
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.platform.ViewRootForTest
+import androidx.compose.ui.semantics.SemanticsNode
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.test.ExperimentalTestApi
+import androidx.compose.ui.test.SemanticsNodeInteraction
+import androidx.compose.ui.test.captureToImage
+import androidx.compose.ui.window.DialogWindowProvider
+import androidx.test.filters.SdkSuppress
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.platform.graphics.HardwareRendererCompat
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import kotlin.math.roundToInt
+
+/**
+ * Capture this [SemanticsNodeInteraction] to an [ImageBitmap], regardless of whether this test runs on Android or the
+ * JVM.
+ *
+ * When running on the JVM, make sure your test class or function has the following annotations. `sdk` has to be >= 26.
+ *
+ * ```kotlin
+ * @GraphicsMode(GraphicsMode.Mode.NATIVE)
+ * @Config(shadows = [ShadowPixelCopy::class], sdk = [26])
+ * ```
+ */
+internal fun SemanticsNodeInteraction.captureToImageCompat(): ImageBitmap =
+    if (System.getProperty("java.runtime.name").orEmpty().lowercase().contains("android")) captureToImage()
+    else captureToImageJvm()
+
+/**
+ * Robolectric-compatible way of capturing a [SemanticsNodeInteraction] to an [ImageBitmap]. It's a copy from
+ * Google's version, minus a forced redraw which doesn't work on Robolectric.
+ *
+ * Make sure your test class or function has the following annotations. `sdk` has to be >= 26.
+ *
+ * ```kotlin
+ * @GraphicsMode(GraphicsMode.Mode.NATIVE)
+ * @Config(shadows = [ShadowPixelCopy::class], sdk = [26])
+ * ```
+ *
+ * See [this issue](https://github.com/robolectric/robolectric/issues/8071) for more info.
+ *
+ * [Source](https://github.com/robolectric/robolectric/issues/8071#issuecomment-1774365222)
+ */
+@OptIn(ExperimentalTestApi::class)
+@SdkSuppress(minSdkVersion = 26)
+private fun SemanticsNodeInteraction.captureToImageJvm(): ImageBitmap {
```

Comment:
Everything below is copied from https://github.com/robolectric/robolectric/issues/8071#issuecomment-1774365222.